sdk/storage/Azure.Storage.Queues/api/Azure.Storage.Queues.netstandard2.0.cs

```diff
@@ -40,24 +40,38 @@ public partial class QueueClient
         public virtual System.Threading.Tasks.Task<Azure.Response> SetAccessPolicyAsync(System.Collections.Generic.IEnumerable<Azure.Storage.Queues.Models.QueueSignedIdentifier> permissions, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.Response SetMetadata(System.Collections.Generic.IDictionary<string, string> metadata, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response> SetMetadataAsync(System.Collections.Generic.IDictionary<string, string> metadata, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.Storage.Queues.Models.UpdateReceipt> UpdateMessage(string messageId, string popReceipt, Azure.Core.BinaryData? message, System.TimeSpan visibilityTimeout = default(System.TimeSpan), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.Response<Azure.Storage.Queues.Models.UpdateReceipt> UpdateMessage(string messageId, string popReceipt, string messageText = null, System.TimeSpan visibilityTimeout = default(System.TimeSpan), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.Storage.Queues.Models.UpdateReceipt>> UpdateMessageAsync(string messageId, string popReceipt, Azure.Core.BinaryData? message, System.TimeSpan visibilityTimeout = default(System.TimeSpan), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.Storage.Queues.Models.UpdateReceipt>> UpdateMessageAsync(string messageId, string popReceipt, string messageText = null, System.TimeSpan visibilityTimeout = default(System.TimeSpan), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
     public partial class QueueClientOptions : Azure.Core.ClientOptions
     {
         public QueueClientOptions(Azure.Storage.Queues.QueueClientOptions.ServiceVersion version = Azure.Storage.Queues.QueueClientOptions.ServiceVersion.V2019_12_12) { }
+        public Azure.Storage.Queues.QueueClientOptions.MessageEncoding Encoding { get { throw null; } set { } }
         public System.Uri GeoRedundantSecondaryUri { get { throw null; } set { } }
         public Azure.Storage.Queues.QueueClientOptions.ServiceVersion Version { get { throw null; } }
+        public enum MessageEncoding
```

Comment:
We should talk with @KrzysztofCwalina about whether or not to nest this enum.  I'd say no, but it matches ServiceVersion which I've also never liked so I may be the wrong person to make that call.  :smile:

Reply:
I've pattern matched ServiceVersion here, so no strong feelings here. taking this out is definitely spare some e-ink.

Reply:
Yeah, I would not nest it. Also, I wonder if we should not move it to Azure.Core (and maybe use it in BinaryData). Lastly, what does None mean? Does it mean "Binary"?

Reply:
`None` means take whatever BinaryData.ToString returns, so maybe we should call it UTF8 ?

Reply:
@JoshLove-msft, what do you think?

Reply:
Regarding DataFormat enum in BinaryData? I think we actually have more of a concrete use case in Event Grid now where the library needs to be able to detect how the BinaryData instance was constructed. Also the Queue use case here is interesting.
One question about this DataFormat - would it be read-only? My interpretation would be that it is meant to determine how the instance was constructed, and users should not be able to set it directly.

Reply:
@JoshLove-msft from Queue POV I'd rather see this as:
`
BinaryData data = ....
string dataInTransit = data.ToString(DataFormat.String);
`
or
`
BinaryData data = ....
string dataInTransit = data.ToString(DataFormat.Base64String);
`

Reply:
I was thinking we would have something like:
```c#
public static BinaryData FromString(string data, DataFormat format = DataFormat.Utf8);

BinaryData base64Data = BinaryData.FromString("data", DataFormat.Base64String);
```

Reply:
Hmm. That would mean customer has to specify data format each time they prepare message - not sure if that would add value for queues as customer could equally use Convert.ToBase64... or so.
For Queues it would rather be that customer gives us bytes and we serialize/deserialize them consistently for given queue instance, so the setting would be 1:1 with queue rather than each message.
(This is coming from the fact that server side accepts only XML friendly string in the request so some transform is needed for other payloads, see related issues: https://github.com/Azure/azure-sdk-for-net/issues/8949 and https://github.com/Azure/azure-sdk-for-net/issues/10242 )

Reply:
So you are saying you just want a way to specify the format on output, but not on input?

Reply:
I see your point about wanting the ability to specify this outside of a particular instance. I think the idea was that the Storage SDK could handle setting the format on the individual BinaryData instances based on this Encoding value that is set in the options. This would work well for output, wouldn't it? Users would just call ToString and the appropriate encoding would be applied. For input, I would think you could use the DataFormat value to determine whether the data needs to be encoded or not:

If MessageEncoding is Base64, and DataFormat is Base64, no additional encoding is needed.
If MessageEncoding is Base64, and DataFormat is default, do Base64 encoding.
If MessageEncoding is None, and DataFormat is default, do nothing.
If MessageEncoding is None, and Dataformat is Base64, decode to Utf-8?

Reply:
The SDK calls ToString before sending payload over the wire [here](https://github.com/kasobol-msft/azure-sdk-for-net/blob/46a008cfd770126fdbd9a854a2164ebcd5797299/sdk/storage/Azure.Storage.Queues/src/QueueClient.cs#L2552-L2581). The data flow is like this Producer Code>BinaryData>SDK API>BinaryData.ToString or .ToBase64String>Rest Call> .... magic (aka cloud) ... > Rest Call > BinaryData.FromString of FromBase64String > SDK API > Consumer Code

I think what you're saying makes sense and we may end up having two switches in play as we can have 2 fairly independent transforms (one in customer's code and one in SDK internals) in both directions.